datafusion/core/src/physical_plan/file_format/file_stream.rs

```diff
@@ -202,30 +205,39 @@ impl<F: FileOpener> FileStream<F> {
         })
     }
 
+    fn next_file(&mut self) -> Option<Result<(FileOpenFuture, Vec<ScalarValue>)>> {
+        let part_file = match self.file_iter.pop_front() {
+            Some(file) => file,
+            None => return None,
+        };
```

Comment:
I think you can use `?` for returning options as well, like

```suggestion
        let part_file =  self.file_iter.pop_front()?;
```